### ISSUE TITLE
Attachments on Spine aren't removed during dress changes

### ISSUE THREAD

**Issue by emmamei** [emmamei/cdkey#154]
An attachment on the Spine is not removed during a dress change - and this is compounded (as expected) by the fact that wear is locked upon change. This is probably related to the Doll Key being locked (on spine) during dress changes.

**Comment by silkymesmeriser** [comment]
They would/should be provided that they are indeed in an RLV folder which also contains an actual worn outfit part (with the exception of parts of the new outfit itself or it's child folders if any) by that of course we are talking about sharing a folder with at least one clothing layer and that folder being under #RLV or alternatively being in the > Outfits folder being used and not being in the new outfits folder or any of it's children.

This assumes the following:
- RLV API implementation client side correctly interprets @detach as a lock on the item issuing the command only, by rights this has always been the meaning of this command even in pre-multiwearable versions of the RLV API that said at the time for most viewers there was no technical distinction between a lock on the item and a lock on the attachment point itself, course this is no longer the case and any recent viewer even one claiming a very old API version if it is still actively maintained should not be conflating the two as they are no longer equivalent.
- Client correctly implements the above.
- Item is not otherwise locked such as by an @detach=n restriction of it's own, or any object or user RLV command issuing a @detach:spine=n,remattach=n,remattach:spine=n,detachthis:/path/to/attachments/folder,detachallthis:,detachallthis:/path/to/attachments/folder/(../)\* or similar restriction issued by any means other than the core scripts built in RLV (UserRLV, Non key scripts, Other objects especially).  If and only if it is being blocked by a restriction from a core script such as those listed which is not being cleared to allow it to proceed then it is a bug.

If it doesn't meet the specified exclusion above then it would sound to me like this is probably actually the result of an expected behaviour and this would be a wontfix or in fact potentially a cantfix where a third party script, object itself or some other third object etc is the issuer of the competing restriction.

Note while it is theoretically possible to parse and filter out any potential restriction that could cause similar from user RLV without regex entries the list of string potentials that would have to be matched against every single user restriction is potentially enormous must be generated internally with every single folder browsed and could take several seconds to match against even a moderately long set of user RLV commands.  For one the singling out of only one attach point to deny being locked in this way is not really justifyable, simply handling then detach:point and remattach:point is a list of 80 restrictions to check for then one @detachall:path=n for every single folder under the outfits tree and one @detachallthis for each too.  FYI in my tiny RLV folder those two would come to about another 80 list items, I can find you examples where that number is multiple thousand each for @detachall and @detachallthis each course those lists would need more than a script of memory each to store.  Further even if it were practical one has to consider the tradeoff in turning off the potential to use the user RLV commands for the purpose of locking important personal items and/or other items that there are completely valid and legitimate reasons to lock in place.

**Comment by emmamei** [comment]
The developer key locks the Key on during dress changes; this was my first thought as to a possible cause. This is what resulted in the obi being kept on even after changing to a different outfit.

This specifically relates to your third bulleted item in the list.

**Comment by silkymesmeriser** [comment]
Oh, your outfit issued @detach=n from within the _outfit_ itself?

Because @detach=n ONLY applies to the object that issued the specific actual @detach=n command and absolutely NOTHING else do not confuse @detach:point=n which applies to an attachment point and @detach=n which applies _ONLY_ to the issuing object since like RLV 1.00 anyway.

**Comment by emmamei** [comment]
As an aside, the current outfit being worn has _no_ clothing parts: it consists of a single attachment and an alpha layer. I wonder if this will be a difficulty, and if it needs a new issue entry.

**Comment by emmamei** [comment]
One thing that OpenCollar does in its most recent incarnation is to offer "fake" clothing for all parts of the body, apparently. Thus, in a situation like this one, one can wear a piece of clothing to be "removed" and the attachments and alpha would then go with it. However, in the case elucidated, it is a _full_ outfit presented as a single attachment. There may not _be_ a good answer to this.

**Comment by silkymesmeriser** [comment]
It might be difficult to find all that many examples of solutions to this sort of situation given that the problem in itself is fairly new and even then has taken time to become particularly common.

The reason being that before rigged mesh the possibility of making a full outfit from a single attachment was basically none existant and even with multiple attachments there would typically be a need for glitchlayers.  Awkward hacks like requiring a layer to add or something might be the only real workaround here.

**Comment by emmamei** [comment]
Closing this as unsolvable - for now.
